std/assembly/polyfills.ts

```diff
@@ -1,8 +1,8 @@
 export function bswap<T>(value: T): T {
-  assert(sizeof<T>() == 1 || sizeof<T>() == 2 || sizeof<T>() == 4 || sizeof<T>() == 8);
+  assert((sizeof<T>() & (sizeof<T>() - 1)) == 0 && sizeof<T>() <= 8);
```

Comment:
couldn't that just `assert(false)` below all the cases?

Reply:
What do you mean?
```
(1 & (1 - 1)) == 0 && 1 <= 8    => true
(2 & (2 - 1)) == 0 && 2 <= 8    => true
(3 & (3 - 1)) == 0 && 3 <= 8    => false
(4 & (4 - 1)) == 0 && 4 <= 8    => true
(5 & (5 - 1)) == 0 && 5 <= 8    => false
(8 & (8 - 1)) == 0 && 8 <= 8    => true
(9 & (9 - 1)) == 0 && 9 <= 8    => false
```

Reply:
I mean that there are already constantly evaluated checks below for 1, 2, 4 and 8 bytes. No matter which check passes, the assert is always there. If you'd move the assert below the 8 bytes case, it could be eliminated in all valid cases.

Reply:
Yeah, you are right!

Reply:
Something like

```ts
export function bswap<T>(value: T): T {
  if (isInteger<T>()) {
    if (sizeof<T>() == 1) {
      return value;
    }
    if (sizeof<T>() == 2) {
      return <T>((value << 8) | ((value >> 8) & <T>0x00FF));
    }
    if (sizeof<T>() == 4) {
      return <T>(
        rotl<u32>(<u32>value & 0xFF00FF00, 8) |
        rotr<u32>(<u32>value & 0x00FF00FF, 8)
      );
    }
    if (sizeof<T>() == 8) {
      let a = (<u64>value >> 8) & 0x00FF00FF00FF00FF;
      let b = (<u64>value & 0x00FF00FF00FF00FF) << 8;
      let v = a | b;
  
      a = (v >> 16) & 0x0000FFFF0000FFFF;
      b = (v & 0x0000FFFF0000FFFF) << 16;
      return <T>rotr<u64>(a | b, 32);
    }
  }
  assert(false);
  return value;
}
```

Reply:
Got it